### PR TITLE
"setHtml" feature, header/footer from pipes and fixed all compiler warnings

### PR DESCRIPTION
**Pull request by hw-dwalter**
This commit adds options for setting HTML source directly!
Also I added some kind of file cache to enable reading from pipes. Pipes can only be read once but until now, header/footer files are read multiple times. If you try to get header/footer from a pipe, it is read the first time while measuring its height. It will read the second time while rendering its content to the PDF, but the pipe is emtpy this time.

In short this patch allows wkhtmltopdf to be called this way (set HTML directly):
```wkhtmltopdf --footer-src "$(cat footer.html)" content.html out.pdf```

Or to call it this way (reading from pipe /dev/fd/xx):
```wkhtmltopdf --footer-html <(cat footer.html) content.html out.pdf```

Some Q_UNUSED and initialisation reordering solved all compiler warnings for me!

### CODE REVIEW

**Comment by michael-iclinic** [comment]
No progress on this pull request?

**Comment by ashkulz** [comment]
@hw-dwalter needs to make the changes as requested above, after which it can be merged.
